causync.py

```diff
@@ -0,0 +1,291 @@
+        """
+
+        cmd = "pgrep -f .*python3.*{}.*{}.*{}".format(self.name, self.src, self.dst)
+
+        try:
+            result = subprocess.check_output(cmd, shell=True).splitlines()
+            self.logger.debug("is_running lines: {}".format(result))
+
+            try:
+                self.pid = result[0].decode()
+            except IndexError:
+                pass
+
+            if len(result) > 1:
+                return True
+            return False
+
+        except subprocess.CalledProcessError as e:
+            # pgrep returns with 1 when there are no results
+            if e.returncode == 1:
+                return False
+            self.logger.debug(("command '{}' returned with error "
+                               "(code {}): {}").format(e.cmd, e.returncode, e.output))
+
+
+
+    def get_logger(self):
+        """ Returns a logger object with the current settings in config.py.
+            If the -q (quiet) flag is set, it doesn't log to console.
+        """
+        logger = logging.getLogger()
+        logger.setLevel(self.config.LOGLEVEL)
+        formatter = logging.Formatter("%(asctime)s %(levelname)s %(message)s")
+
+        # logfile handler
+        lf_handler = logging.FileHandler(self.config.LOGFILE)
+        lf_handler.setFormatter(formatter)
+        logger.addHandler(lf_handler)
+
+        if not self.quiet:
+            # console handler
+            c_handler = logging.StreamHandler()
+            c_handler.setFormatter(formatter)
+            logger.addHandler(c_handler)
+
+        return logger
+
+
+def parse_args():
+    """ Parses command-line arguments and
+        sets a few variables depending on the 'task' argument.
+    """
+    parser = ArgumentParser(description="Causality backup solution")
+
+    parser.add_argument('task', choices=['check', 'sync', 'cleanup'], help='task to execute')
+
+    parser.add_argument('source', help='sync source directory')
+
+    parser.add_argument('destination', help='sync destination directory')
+
+    parser.add_argument('--incremental',
+                        dest='incremental',
+                        action='store_true',
+                        default=False,
+                        help='incremental backup')
+
+    parser.add_argument('--incremental-basedir',
+                        dest='incremental_basedir',
+                        action='store',
+                        default=None,
+                        help='basedir of full backup')
+
+    parser.add_argument('-q',
+                        '--quiet',
+                        action='store_true',
+                        default=False,
+                        help="don't print to console")
+
+    arguments = parser.parse_args()
+
+    arguments.selfname = sys.argv[0]
+    arguments.checkonly = True if arguments.task == 'check' else False
+    arguments.cleanup = True if arguments.task == 'cleanup' else False
+
+    if arguments.incremental and not arguments.incremental_basedir:
```

Comment:
Incremental backup would be nice to have in an automatic way if previous backup directories exist.